### PR TITLE
fix: count severity by product [HEAD-879]

### PR DESCRIPTION
**Pull request by cat2608**
### Description

This PR counts issue's severities per product to enhance analytics and reporting.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: domain/ide/workspace/folder.go [hunk 1]
```diff
@@ -181,8 +181,10 @@ func (f *Folder) DocumentDiagnosticsFromCache(file string) []snyk.Issue {
 func (f *Folder) processResults(scanData snyk.ScanData) {
 	if scanData.Err != nil {
 		f.scanNotifier.SendError(scanData.Product, f.path)
-		log.Err(scanData.Err).Str("method", "processResults").
-			Msgf("%s returned an error: %s", scanData.Product, scanData.Err.Error())
+		log.Err(scanData.Err).
+			Str("method", "processResults").
+			Str("product", string(scanData.Product)).
+			Msg("Product returned an error")
```
Comment (cat2608) [inline domain/ide/workspace/folder.go:187]: 💅 Cosmetic change: `Msgf` says:

>NOTICE: once this method is called, the *Event should be disposed. Calling Msgf twice can have unexpected result.

updating-log

File: domain/ide/workspace/folder.go [hunk 2]
```diff
@@ -275,14 +281,15 @@ func (f *Folder) FilterAndPublishCachedDiagnostics(product product.Product) {
...
+	logger := log.With().Str("method", "filterCachedDiagnostics").Logger()
 
 	var issuesByFile = map[string][]snyk.Issue{}
 	if f.documentDiagnosticCache.Size() == 0 {
 		return issuesByFile
 	}
 
-	logger.Debug().Msgf("Filtering issues by severity: %v", config.CurrentConfig().FilterSeverity())
+	filterSeverity := config.CurrentConfig().FilterSeverity()
+	logger.Debug().Interface("filterSeverity", filterSeverity).Msg("Filtering issues by severity")
```
Comment (cat2608) [inline domain/ide/workspace/folder.go:292]: 💅 Cosmetic change:

From:
```bash
2023-11-01T15:18:59Z DBG processResults - Filtering issues by severity: {true true true true}|
```
To:
```bash
2023-11-01T15:24:48Z DBG filterCachedDiagnostics - Filtering issues by severity filterSeverity={"critical":true, "high":true,"low": true, "medium": true}
```

File: domain/ide/workspace/folder_test.go [hunk 3]
```diff
@@ -507,6 +507,45 @@ func Test_processResults_ShouldNotSendAnalyticsToAPIIfDisabled(t *testing.T) {
...
+	engineMock.EXPECT().InvokeWithInputAndConfig(localworkflows.WORKFLOWID_REPORT_ANALYTICS, gomock.Any(), gomock.Any()).Times(0)
+
+	// Act
+	f.processResults(scanData1)
+	// f.processResults(scanData2)
+
+	// Assert
+	if got, want := scanData1.SeverityCount[product.ProductOpenSource][snyk.Critical.String()], 2; got != want {
+		t.Errorf("got %v, want %v", got, want)
+	}
```
Comment (cat2608) [inline domain/ide/workspace/folder_test.go:546]: WIP: I'm not sure how to assert `scanData`... 
Change `processResults` to accept `scanData` as a pointer? 🤔

Comment (bastiandoetsch) [reply domain/ide/workspace/folder_test.go:546]: suggestion: how about `require.Equal(t, 2, scanData1.SeverityCount[product.ProductOpenSource][snyk.Critical.String()])`? Would be easier for me to understand. Not sure about your question, calling :)

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**